### PR TITLE
Console - Migrate some pages to material MDC

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue
n/a
## Description

Migrate : api audit events, api cors, api get started, api creation V4, api deployment config v4, api documentation V4, api endpoints V2 & V4

## Additional context

🏗️ Your changes can be tested here and will be available soon:
      Console: [https://pr.team-apim.gravitee.dev/6618/console](https://pr.team-apim.gravitee.dev/6618/console)
      Portal: [https://pr.team-apim.gravitee.dev/6618/portal](https://pr.team-apim.gravitee.dev/6618/portal)
      Management-api: [https://pr.team-apim.gravitee.dev/6618/api/management](https://pr.team-apim.gravitee.dev/6618/api/management)
      Gateway v4: [https://pr.team-apim.gravitee.dev/6618](https://pr.team-apim.gravitee.dev/6618)
      Gateway v3: [https://pr.gateway-v3.team-apim.gravitee.dev/6618](https://pr.gateway-v3.team-apim.gravitee.dev/6618)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-yalpvzakfi.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/creation-get-started/api-creation-get-started.component.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@
 import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
 import { catchError, filter, switchMap, takeUntil, tap } from 'rxjs/operators';
 import { EMPTY, of, Subject } from 'rxjs';
-import { MatLegacyDialog as MatDialog } from '@angular/material/legacy-dialog';
+import { MatDialog } from '@angular/material/dialog';
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/creation-get-started/api-creation-get-started.component.ts:20]: here we use a MatDialog but it line 87 it reference a component that has not been migrated. So the import is broken if we don't migrate both components. Can we do it in this PR before merging ?

File: gravitee-apim-console-webui/src/management/api/component/gio-connector-dialog/gio-connector-dialog.component.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 import { Component, Inject } from '@angular/core';
-import { MatLegacyDialogRef as MatDialogRef, MAT_LEGACY_DIALOG_DATA as MAT_DIALOG_DATA } from '@angular/material/legacy-dialog';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/component/gio-connector-dialog/gio-connector-dialog.component.ts:17]: we broken the design here I think. Maybe we just need to add some padding in the dialog. ( I'm thinking about the button at the right top corner )
Capture d’écran 2024-02-15 à 16 06 17
Capture d’écran 2024-02-15 à 16 06 41

File: gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.component.html [hunk 3]
```diff
@@ -16,85 +16,87 @@
...
-        </button>
+          <button
+            id="switchListenerType"
+            mat-button
+            type="button"
+            (click)="this.enableVirtualHost = !this.enableVirtualHost"
+            [disabled]="this.enableVirtualHost && this.domainRestrictions.length > 0"
+          >
+            <ng-container *ngIf="!this.enableVirtualHost"> <mat-icon svgIcon="gio:check"></mat-icon> Enable virtual hosts </ng-container>
+            <ng-container *ngIf="this.enableVirtualHost"> <mat-icon svgIcon="gio:cancel"></mat-icon> Disable virtual hosts </ng-container>
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.component.html]: The ng-container break the design I think. We should change for something like this
```
    <button
            id="switchListenerType"
            mat-button
            type="button"
            (click)="this.enableVirtualHost = !this.enableVirtualHost"
            [disabled]="this.enableVirtualHost && this.domainRestrictions.length > 0"
          >
            <mat-icon [svgIcon]="enableVirtualHost ? 'gio:cancel' : 'gio:check'"></mat-icon>
            {{ enableVirtualHost ? 'Disable virtual hosts' : 'Enable virtual hosts' }}
          </button>
```

Capture d’écran 2024-02-15 à 16 29 26

File: gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-5-summary/step-5-summary.component.html [hunk 4]
```diff
@@ -16,166 +16,168 @@
...
-      >
-        Save API & Ask for a review<mat-icon
-          svgIcon="gio:info"
-          class="step-5-summary__footer__button-icon"
-          matTooltip="Your API definition will be saved and asked for review."
-        ></mat-icon>
-      </button>
+    <div class="step-5-summary__footer">
+      <div class="step-5-summary__footer__actions">
+        <button mat-stroked-button type="button" (click)="createApi({ deploy: false, askForReview: false })">
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-5-summary/step-5-summary.component.html:139]: The design is broken here we miss some margin on the icon I think.

Capture d’écran 2024-02-15 à 16 32 37

File: gravitee-apim-console-webui/src/management/api/documentation-v4/dialog/documentation-edit-folder-dialog/api-documentation-v4-edit-folder-dialog.component.scss
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/documentation-v4/dialog/documentation-edit-folder-dialog/api-documentation-v4-edit-folder-dialog.component.scss:1]: We need to add a div and some padding on this dialog I think

Capture d’écran 2024-02-15 à 16 39 18

File: gravitee-apim-console-webui/src/management/api/component/gio-connector-dialog/gio-connector-dialog.component.ts (hunk 2 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/component/gio-connector-dialog/gio-connector-dialog.component.ts:17]: I'll keep it that way. It's a similar display on all the dialogs and a subject to fix globally with a single component. IMHO

File: gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-5-summary/step-5-summary.component.html (hunk 4 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-5-summary/step-5-summary.component.html:139]: good catch. 
Like all component override, the best thing to do is 🚮 the css

File: gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.component.html (hunk 3 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.component.html]: 👌

File: gravitee-apim-console-webui/src/management/api/creation-get-started/api-creation-get-started.component.ts (hunk 1 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/creation-get-started/api-creation-get-started.component.ts:20]: i will migrate it later. I've change to keep legacy dialog here

**Review by Okhelifi — APPROVED** [review]

**Pull request merged**